stellargraph/mapper/full_batch_generators.py

```diff
@@ -503,3 +504,30 @@ def flow(self, node_ids, targets=None):
         return RelationalFullBatchNodeSequence(
             self.features, self.As, self.use_sparse, targets, node_indices
         )
+
+
+def corrupt_full_batch_node_generator(generator):
+    """
+    This functions takes FullBatchNodeGenerator object and returns a Sequence object for training a full batch
+    Deep Graph Infomax model.
+
+    Args:
+        generator (FullBatchNodeGenerator): the node generator object
+    Returns:
+        a Sequence object for Deep Graph Infomax training
+    """
+
+    if not isinstance(generator, FullBatchNodeGenerator):
+        raise TypeError(
+            f"generator: expected FullBatchNodeGenerator, found {type(generator).__name__}"
+        )
+
+    num_nodes = generator.features.shape[0]
+    node_indices = np.arange(num_nodes)
+    targets = np.zeros((num_nodes, 2))
+    targets[:, 0] = 1.0
```

Comment:
What are `targets` used for?

Reply:
The mode outputs scores (w /out sigmoid applied) with shape (1 x N x 2), where the first column correpsonds to true nodes and the second column corresponds to corresponds to corrupted nodes. Used `targets`, the model output and `targets` can be passed to `tf.nn.sigmoid_cross_entropy_with_logits` to compute the Deep Graph Infomax loss function